docs/design/persistent-storage.md

```diff
@@ -0,0 +1,220 @@
+
+Susie is happy to see storage that can be mounted many times (ReadOnlyMany).  That satisfies her load balancer requirement.  Susie creates an pod image with her number crunching code and requests persistence for her volume.
+
+Susie's YAML looks something like this for the pod that creates her data:
+
+
+```yaml
+
+id: creator-of-data
+kind: Pod
+namespace: ns
+apiVersion: v1beta1
+desiredState:
+  manifest:
+    version: v1beta1
+    id: creator-of-data
+    containers:
+      - name: numbercruncher
+        image: ns/makesomdata:v1
+        command:
+          - /run.sh
+        volumeMounts:
+          - name: data
+            mountPath: /some_data
+    volumes:
+      - name: data
+        persistenceRequirements:
+        	mode: readonlymany
+        	retentionPolicy: retainonpodexit
+        	resources:
+        		- size: 5
+        	selector:
+        		- name: my-persistent-data
+labels:
+  name: creator-of-data
+
+```
+
+#### Behind the curtain
+
+`PersistentStorageBinder` is watching for new pods with `PersistenceRequirements`.  The binder compares the request to all available storage and binds it to the closest match.
+
+To bind:
+
+* Set PersistentStorage.Status.PodRef = pod that originally requested storage
+* Set PersistentStorage.Labels = PersistenceRequirements.Selector
+
+The scheduler will wait to schedule a pod until its storage requirements are fulfilled.  Invert responsibility for host selection when scheduling pods with persistent storage.  Allow the controller to give advice to the scheduler, keeping storage rules contained in the provider. 
+
+Storage controllers are watching bound pods to attach volumes for cloud providers.
+
+Kubelet waits to mount a filesystem until a cloud provider has finished attaching the volume.
+
+
+#### Surviving the pod lifecycle
+
+Susie's pod finishes its work and exits.  Because her retention policy was "RetainOnPodExit", she is relying on Kubernetes to make the data available again to her in the same namespace using the same selector.
+
+Susie can now use a ReplicaController to create many pods to serve her content.  Because Susie is a responsible kubitzen (a kube citizen?), she set the retention policy to DeleteOnPodExit, so that when all her replicas are done serving content, the persistent storage object can be recycled back into the pool.
+
+
+```
+id: server-of-data
+kind: ReplicationController
+apiVersion: v1beta1
+desiredState:
+  replicas: 10
+  replicaSelector:
+    name: server-of-data
+  podTemplate:
+    desiredState:
+      manifest:
+        version: v1beta1
+        id: server-of-data
+        containers:
+          - name: server-of-data
+            image: ns/servesomedata:v1
+            command:
+              - /run.sh
+            volumeMounts:
+              - name: data
+                mountPath: /some_data
+        volumes:
+      - name: data
+        persistenceRequirements:
+        	mode: readonlymany
+        	retentionPolicy: deleteonpodexit
```

Comment:
I think mixing deleteOnPodExit with a replication controller would be a very dangerous idea. Even if it's smart enough to wait until the last pod exits (i.e. there aren't any mounts), it's possible the that replication controller itself is still active and trying to create more pods. If it's too slow making a new one, the storage could be gone before it succeeds and the pod would be left without its data. It seems like a deleteOnReplicationControllerDeletion policy is what would really be needed.